### ISSUE TITLE
Cannnot reproduce the results

### ISSUE THREAD

**Issue by addiu** [microsoft/UST#7]
Hi, thanks a lot for providing the code for the nice paper.
I tried this code for the SST-2 and the AG-News dataset, I followed all parameter settings listed in the paper's appendix. However, I am not able to reproduce the result.
More precise, I observed the following two issues:
1. the BERT base model performs actually very well given 30 labels. I always achieve an accuracy of  80%+ for SST-2 and 85%+ for AG-News. Yet the paper reports 69.79% and 80.74% for them. I already tried different seeds but observe the same results.
2. I haven't observed much improvement after using UST on unlabeled data, all within 1%.  Hence, I can not reproduce the results given in the paper.

As an example: I run this command and got the my observed SST-2 results mentioned above:

CUDA_VISIBLE_DEVICES=$cuda_id PYTHONHASHSEED=42 python ../run_ust.py \
--task /local/data \
--model_dir /local/sst-2 \
--seq_len 32 \
--sup_batch_size 4 \
--unsup_batch_size 32 \
--sample_size 16384 \
--unsup_size 4096 \
--sample_scheme easy_bald_class_conf \
--sup_labels 60 \
--T 30 \
--alpha 0.1 \
--valid_split 0.5 \
--sup_epochs 50 \
--unsup_epochs 25 \
--N_base 5 \
--pt_teacher TFBertModel \
--pt_teacher_checkpoint bert-base-uncased \
--hidden_dropout_prob 0.3 \
--attention_probs_dropout_prob 0.3 \
--dense_dropout 0.5

My questions:
1. Is there something we need to take care of when creating the datasets (preprocessing)?
2. there is a hyperparameter, called alpha, in the code. I think this hyperparameter is not mentioned in the paper. Shall I use the default value (i.e. 0.1) when running the code?

**Comment by igorhuangchao** [comment]
@microsoftopensource @subhomj thanks

**Comment by subhomj** [comment]
For reproducibility:

1. Use the original code used in the paper (prior to HuggingFace integration) arxived in NeurIPS website [https://proceedings.neurips.cc/paper/2020/file/f23d125da1e29e34c552f448610ff25f-Supplemental.zip](https://proceedings.neurips.cc/paper/2020/file/f23d125da1e29e34c552f448610ff25f-Supplemental.zip). The fine-tuning difference could be due to the underlying DL package.

2. For improved UST performance, refer to the ReadME section on "Continued Pre-training on Task-specific Unlabeled Data". Essentially, you can initialize the pt_teacher_checkpoint with the continued pre-trained checkpoint on unlabeled data for the specific task. These checkpoints are available in the directory above.
